scripts/deno_bridge.ts

```diff
@@ -0,0 +1,192 @@
+            if (mod_names.indexOf(ctx.name) === -1) {
+                mod_names.push(ctx.name)
+            }
+        } else if (ctx.type === 'from') {
+            for (var j = 0, _len_j = ctx.names.length; j < _len_j; j++) {
+                var name = ctx.names[j];
+                if (name === '*') {
+                    // just pass, we don't want to include '*'
+                } else if (ctx.aliases[name] !== undefined) {
+                    if (mod_names.indexOf(ctx.aliases[name]) === -1) {
+                        mod_names.push(ctx.aliases[name])
+                    }
+                } else {
+                    if (mod_names.indexOf(ctx.names[j]) === -1) {
+                        mod_names.push(ctx.names[j])
+                    }
+                }
+            }
+        } else if (ctx.type === 'assign') {
+            var left = ctx.tree[0]
+            if (left.type === 'expr' && left.tree[0].type === 'id' && left.tree[0].tree.length === 0) {
+                var id_name = left.tree[0].value
+                if (mod_names.indexOf(id_name) === -1) {
+                    mod_names.push(id_name)
+                }
+            }
+        }
+    }
+    // create the object that will be returned when the anonymous function is run
+    var ret_code = 'return {'
+    for (var i = 0, _len_i = mod_names.length; i < _len_i; i++) {
+        ret_code += mod_names[i] + ':' + mod_names[i] + ','
+    }
+    ret_code += '__getattr__:function(attr){return this[attr]},'
+    ret_code += '__setattr__:function(attr,value){this[attr]=value}'
+    ret_code += '}'
+    // @ts-ignore
+    var ret_node = new $Node('expression')
+    // @ts-ignore
+    new $NodeJSCtx(ret_node, ret_code)
+    mod_node.add(ret_node)
+    // add parenthesis for anonymous function execution
+
+    // @ts-ignore
+    var ex_node = new $Node('expression')
+    // @ts-ignore
+    new $NodeJSCtx(ex_node, ')()')
+    root.add(ex_node)
+
+    try {
+        var js = root.to_js()
+        return js;
+    } catch (err) {
+        eval('throw ' + err.name + '(err.message)')
+    }
+    return undefined;
+}
+
+async function execute_python_script(filename: string) {
+    const _py_src = await fs.readFileStr(filename, { encoding: 'utf8' })
+    // @ts-ignore
+    __BRYTHON__.$py_module_path['__main__'] = './'
+    // @ts-ignore
+    var js = __BRYTHON__.python_to_js(_py_src, '__main_soln__')
+    eval(js);
+}
+
+// @ts-ignore
+__BRYTHON__.$py_module_path = __BRYTHON__.$py_module_path || {}
+// @ts-ignore
+__BRYTHON__.$py_module_alias = __BRYTHON__.$py_module_alias || {}
+// @ts-ignore
+__BRYTHON__.exception_stack = __BRYTHON__.exception_stack || []
+// @ts-ignore
+__BRYTHON__.scope = __BRYTHON__.scope || {}
+// @ts-ignore
+__BRYTHON__.imported = __BRYTHON__.imported || {}
+// @ts-ignore
+__BRYTHON__.modules = __BRYTHON__.modules || {}
+// @ts-ignore
+__BRYTHON__.compile_python = $compile_python
+// @ts-ignore
+__BRYTHON__.import_hooks = $import_hooks
```

Comment:
I see that $import_hooks was declared to be an async function, which means that it returns a Promise. I'm surprised that that would work with Brython, which expects __BRYTHON__.import_hooks to be a regular function. Are you sure that it works?

Reply:
Whole this implementation is just first step to support Deno ... currently it is not working (

Reply:
OK, I didn't realize this was still a work in progress. I'm inclined to close the pull request, then, until you've got it working. What do you think?